akka-actor/src/main/scala/akka/actor/dungeon/FaultHandling.scala

```diff
@@ -49,6 +50,7 @@ private[akka] trait FaultHandling { this: ActorCell =>
   private def setFailed(perpetrator: ActorRef): Unit = _failed = perpetrator
   private def clearFailed(): Unit = _failed = null
   private def perpetrator: ActorRef = _failed
+  var _failedFatally: Boolean = false
```

Comment:
What's the impact on memory size?

Reply:
Another possible solution to having this as a separate field: you could wrap the fatal failure in a private "FatalActorFailure(…, cause)" and test against that and pass on the getCause. This would only incur overhead in the fatal failure case, and it could have with NoStackTrace to minimize runtime costs.

Reply:
Thanks, I had not looked after alternatives to carry that flag, the exception wrapper might be fine.

Reply:
:+1:

Reply:
I don't think it's easy to carry it in a special exception because there can be user defined supervision inbetween, and probably also complicated for remote deployment.

We could probably use the `_failed` field to hold more than just an `ActorRef`.

Reply:
@patriknw You mean that it could be possible to not receive the same exception back from the supervisor?

Reply:
yes, I have updated this to encode that flag in the existing field: 44433b9

Reply:
@patriknw :+1: